File: ATB_1.1_src/src/main/java/edu/stanford/isis/atb/domain/template/Inference.java
/*L
*  Copyright Northwestern University
*  Copyright Stanford University (ATB 1.0 and ATS 1.0)
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/



package edu.stanford.isis.atb.domain.template;

import javax.validation.Valid;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;
import org.simpleframework.xml.core.Commit;

/**
 * @author Vitaliy Semeshko
 */

@Root(name="Inference", strict=false)
public class Inference extends AbstractRemovableElement {

	@Attribute
	private boolean annotatorConfidence;
	
	@Valid
	@Element(name="QuestionType", required=false)
	private QuestionType questionType;
	
	@SuppressWarnings("unused")
	@Commit
	private void afterPropertiesSet() {
		if (questionType != null) {
			questionType.setRemoveCommand(new RemoveCommand() {
				@Override
				public void execute() {
					questionType = null;
				}
			});
		}
	}

	public boolean isAnnotatorConfidence() {
		return annotatorConfidence;
	}

	public void setAnnotatorConfidence(boolean annotatorConfidence) {
		this.annotatorConfidence = annotatorConfidence;
	}

	public QuestionType getQuestionType() {
		return questionType;
	}

	public void setQuestionType(QuestionType questionType) {
		this.questionType = questionType;
		questionType.setRemoveCommand(new RemoveCommand() {
			@Override
			public void execute() {
				Inference.this.questionType = null;
			}
		});
	}
	
	@Override
	public String toString() {
		return "Inference";
	}
	
}

File: AIMToolkit_v3.0.2_rv11/source/AIMLib/AIMLib/model/Inference.cpp
/*
*  2007 – 2013 Copyright Northwestern University
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/


#include "../stdafx.h"

#include "Inference.h"

#include "../memdebug.h"

using namespace aim_lib;

Inference::Inference(void)
{
	_imageEvidence = true;
}

Inference::Inference(const Inference& inference)
{
	*this = inference;
}

Inference::~Inference(void)
{
}

Inference& Inference::operator= (const Inference& inference){
	_cagridId				= inference._cagridId;
	_codeValue				= inference._codeValue;
	_codeMeaning			= inference._codeMeaning;
	_codingSchemeDesignator	= inference._codingSchemeDesignator;
	_codingSchemeVersion	= inference._codingSchemeVersion;
	_imageEvidence			= inference._imageEvidence;

	SetAnnotatorConfidence(inference.GetAnnotatorConfidence());

	return *this;
}

const std::string& Inference::GetCodeMeaning() const{

	return _codeMeaning;
}

const std::string& Inference::GetCodeValue() const{

	return _codeValue;
}

const std::string& Inference::GetCodingSchemeDesignator() const{

	return _codingSchemeDesignator;
}

const std::string& Inference::GetCodingSchemeVersion() const{

	return _codingSchemeVersion;
}

double* Inference::GetAnnotatorConfidence() const{
	return _annotatorConfidence.get();
}

bool Inference::GetImageEvidence() const{
	return _imageEvidence;
}

void Inference::SetCodeMeaning(const std::string& newVal){

	_codeMeaning = newVal;
}

void Inference::SetCodeValue(const std::string& newVal){

	_codeValue = newVal;
}

void Inference::SetCodingSchemeDesignator(const std::string& newVal){

	_codingSchemeDesignator = newVal;
}

void Inference::SetCodingSchemeVersion(const std::string& newVal){

	_codingSchemeVersion = newVal;
}

void Inference::SetAnnotatorConfidence(double* pConfidence){

	if (pConfidence == NULL)
		_annotatorConfidence.reset();
	else
		_annotatorConfidence.reset(new double(*pConfidence));
}

void Inference::SetImageEvidence(bool newVal){

	_imageEvidence = newVal;
}
